### PR TITLE
Update reason for appointment for community care

### PR DESCRIPTION
**Pull request by narin**
## Description
Updates page title and ui schema on Reason For Appointment page if CC.  Radio buttons should be hidden, question copy should be different, and textarea should be optional.

This also fixes a bug on the CC preferences page where the city, state was showing up as `undefined, undefined` on localhost.  I found that this was because the selector for city, state was incorrect in the reducer.  Also fixed the transformer for Organization.  The address field should be an array of objects instead of an object.

## Testing done
Local and unit

## Screenshots
image

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/vaos/reducers/newAppointment.js [hunk 1]
```diff
@@ -578,18 +582,34 @@ export default function formReducer(state = initialState, action) {
...
 
+      let additionalInfoTitle;
+
+      // If community care, hide reasonForAppointment radios, update
+      // additional info title, remove expand condition and make optional
+      if (isCommunityCare) {
+        additionalInfoTitle = REASON_ADDITIONAL_INFO_TITLES.cc;
+        reasonSchema = unset('properties.reasonForAppointment', reasonSchema);
+        reasonSchema = unset('required', reasonSchema);
+        delete uiSchema.reasonAdditionalInfo['ui:options'].expandUnder;
```
Comment (jbalboni) [inline src/applications/vaos/reducers/newAppointment.js:593]: It's pretty unlikely, but I believe with this change, if you go through the CC flow, then go all the way back start back through the VA flow, this page won't work. You're making a shallow copy on line 566, but these are deleting data in sub-objects, which aren't copied. I could be wrong, definitely verify that before making any changes.

If that is true, you could change these deletes to `unset`, that might fix it. You could also have two different versions of the uiSchema in the page component and have the reducer switch between them depending on the flow.

File: src/applications/vaos/services/organization/transformers.js [hunk 2]
```diff
@@ -31,12 +31,14 @@ export function transformParentFacilities(parentFacilities) {
       ],
       active: true,
       name: facility.authoritativeName,
-      address: {
-        line: [],
-        city: facility.city,
-        state: facility.stateAbbrev,
-        postalCode: null,
-      },
+      address: [
```
Comment (jbalboni) [inline src/applications/vaos/services/organization/transformers.js:34]: Do we use this address anywhere else? I thought we did on the facility page, but looks like we don't actually.

File: src/applications/vaos/reducers/newAppointment.js (hunk 1 above)
Comment (narin) [reply src/applications/vaos/reducers/newAppointment.js:593]: I tested going back and it seemed fine, but created separate schemas anyway

File: src/applications/vaos/services/organization/transformers.js (hunk 2 above)
Comment (narin) [reply src/applications/vaos/services/organization/transformers.js:34]: I couldn't find anywhere else we are using the Organization/system address besides the CC preferences page.

File: src/applications/vaos/containers/ReasonForAppointmentPage.jsx [hunk 3]
```diff
@@ -11,51 +11,92 @@ import SchemaForm from 'platform/forms-system/src/js/components/SchemaForm';
...
 const pageKey = 'reasonForAppointment';
-const pageTitle = 'Choose a reason for your appointment';
 
 export class ReasonForAppointmentPage extends React.Component {
   componentDidMount() {
-    document.title = `${pageTitle} | Veterans Affairs`;
-    this.props.openReasonForAppointment(pageKey, uiSchema, initialSchema);
+    const isCommunityCare = this.isCommunityCare();
+    const reasonUiSchema = isCommunityCare ? uiSchema.cc : uiSchema.default;
+    const reasonInitialSchema = isCommunityCare
```
Comment (jbalboni) [inline src/applications/vaos/containers/ReasonForAppointmentPage.jsx:78]: You could choose the schema in a constructor, that would avoid some of the checking you have to do in the `render` method.

File: src/applications/vaos/reducers/newAppointment.js [hunk 4]
```diff
@@ -562,11 +563,13 @@ export default function formReducer(state = initialState, action) {
       };
     }
     case FORM_REASON_FOR_APPOINTMENT_PAGE_OPENED: {
+      const uiSchema = { ...action.uiSchema };
```
Comment (jbalboni) [inline src/applications/vaos/reducers/newAppointment.js:566]: I don't think you need to copy this now.

**Review by jbalboni — APPROVED** [review]
Some minor comments, but LGTM overall

File: src/applications/vaos/containers/ReasonForAppointmentPage.jsx [hunk 5]
```diff
@@ -11,51 +11,92 @@ import SchemaForm from 'platform/forms-system/src/js/components/SchemaForm';
...
+        type: 'string',
+      },
     },
-    reasonAdditionalInfo: {
-      type: 'string',
+  },
+  cc: {
+    type: 'object',
+    properties: {
+      reasonAdditionalInfo: {
```
[the remark this answers is not in the document]
Comment (narin) [reply src/applications/vaos/containers/ReasonForAppointmentPage.jsx:36]: Nope, it's optional for CC

[the remark this answers is not in the document]
Comment (jbalboni) [reply src/applications/vaos/containers/ReasonForAppointmentPage.jsx:36]: Ok, in that case we might want to tweak the logic in the submitAppointmentOrRequest action. It assumes there's always a request message to send.

File: src/applications/vaos/actions/newAppointment.js [hunk 6]
```diff
@@ -754,10 +754,10 @@ export function submitAppointmentOrRequest(router) {
...
 
         try {
-          await sendRequestMessage(
-            requestData.id,
-            newAppointment.data.reasonAdditionalInfo,
-          );
+          const requestMessage = newAppointment.data.reasonAdditionalInfo;
+          if (requestMessage) {
+            await sendRequestMessage(requestData.id, requestMessage);
+          }
```
Comment (narin) [inline src/applications/vaos/actions/newAppointment.js:760]: Added check for `reasonAdditionalInfo` before sending message

**Pull request merged**